From shimoda.yoshihiro@renesas.com Wed May 30 04:42:52 2007
From: Yoshihiro Shimoda <shimoda.yoshihiro@renesas.com>
Date: Wed, 30 May 2007 20:42:41 +0900
Subject: USB: r8a66597-hcd: fix NULL access
To: greg@kroah.com, linux-usb-devel@lists.sourceforge.net
Message-ID: <465D6331.9070302@renesas.com>


This patch fixes the problem that accesses NULL pointer
when disconnected a cable while play music with usb-speaker.

Signed-off-by: Yoshihiro Shimoda <shimoda.yoshihiro@renesas.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/r8a66597-hcd.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/usb/host/r8a66597-hcd.c
+++ b/drivers/usb/host/r8a66597-hcd.c
@@ -47,7 +47,7 @@ MODULE_DESCRIPTION("R8A66597 USB Host Co
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Yoshihiro Shimoda");
 
-#define DRIVER_VERSION	"9 May 2007"
+#define DRIVER_VERSION	"29 May 2007"
 
 static const char hcd_name[] = "r8a66597_hcd";
 
@@ -544,6 +544,9 @@ static void pipe_toggle_restore(struct r
 	unsigned char endpoint = usb_pipeendpoint(urb->pipe);
 	unsigned short *toggle = get_toggle_pointer(dev, urb->pipe);
 
+	if (!toggle)
+		return;
+
 	r8a66597_pipe_toggle(r8a66597, pipe, *toggle & (1 << endpoint));
 }
 
